daml-lf/spec/daml-lf-1.rst

```diff
@@ -2431,605 +2380,206 @@ exact output.
 ::
 
   Evaluation result
-    r ::= Ok v                                      -- ResOk
-       |  Err E                                     -- ResErr
-    E ::= Throw v                                   -- ErrThrow, v is a value of AnyException type
-       |  Fatal t                                   -- ErrFatal, t is a text value
-
-                           ┌──────────┐
-  Big-step evaluation      │ e  ⇓  r  │
-                           └──────────┘
-
-    —————————————————————————————————————————————————————————————————————— EvValue
-      v  ⇓  Ok v
-
-
-      e   ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvTyAbsErasableErr
-      Λ α : ek . e  ⇓  Err E
-
-
-      e   ⇓  Ok v
-    —————————————————————————————————————————————————————————————————————— EvTyAbsErasable
-      Λ α : ek . e  ⇓  Ok (Λ α : ek . v)
 
+    r ::= Ok e                                      -- ResOk
+       |  Err err                                   -- ResErr
+    err ::=  Throw v                                -- ErrThrow, v is a value of AnyException type
+         |   Fatal t                                -- ErrFatal, t is a text value
 
-      e₁  ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvExpAppErr1
-      e₁ e₂  ⇓  Err E
+                                              ┌──────┐
+  Applying result to an evaluation context    │ E[r] │
+                                              └──────┘
 
-      e₁  ⇓  Ok (λ x : τ . e)
-      e₂  ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvExpAppErr2
-      e₁ e₂  ⇓  Err E
+    E[ Ok e ] = Ok E[e]
+    E[ Err err ] = Err err
```

Comment:
I would like to avoid using equality in an operational semantics.
One possibility is to make those statements meta, saying something like  "in the following we use `E[r]` where `r` is a result as syntactic sugar for `OK E[ e ]` if `r` is a success result of the form `OK e` or  `Err e` if `r` is a result of the form `Err r`. 
The other possibility, I think is nicer is to defined all your rule on redex only (i.e. with any context), and adding two rules for contexts (see below).